sdk/src/assertions/ingredient.rs

```diff
@@ -40,33 +43,41 @@ pub enum Relationship {
 }
 
 /// An ingredient assertion
-#[derive(Serialize, Deserialize, Debug, Default)]
+#[derive(Debug, Default, PartialEq)]
 pub struct Ingredient {
-    #[serde(rename = "dc:title")]
-    pub title: String,
-    #[serde(rename = "dc:format")]
-    pub format: String,
-    #[serde(rename = "documentID", skip_serializing_if = "Option::is_none")]
+    pub title: Option<String>,
+    pub format: Option<String>,
     pub document_id: Option<String>,
-    #[serde(rename = "instanceID", skip_serializing_if = "Option::is_none")]
     pub instance_id: Option<String>,
-    #[serde(skip_serializing_if = "Option::is_none")]
     pub c2pa_manifest: Option<HashedUri>,
-    #[serde(rename = "validationStatus", skip_serializing_if = "Option::is_none")]
     pub validation_status: Option<Vec<ValidationStatus>>,
     pub relationship: Relationship,
-    #[serde(skip_serializing_if = "Option::is_none")]
     pub thumbnail: Option<HashedUri>,
-    #[serde(skip_serializing_if = "Option::is_none")]
     pub metadata: Option<Metadata>,
-    #[serde(skip_serializing_if = "Option::is_none")]
     pub data: Option<HashedUri>,
-    #[serde(skip_serializing_if = "Option::is_none")]
     pub description: Option<String>,
-    #[serde(rename = "informational_URI", skip_serializing_if = "Option::is_none")]
     pub informational_uri: Option<String>,
-    #[serde(skip_serializing_if = "Option::is_none")]
     pub data_types: Option<Vec<AssetType>>,
+
+    pub validation_results: Option<ValidationResults>,
+    pub active_manifest: Option<HashedUri>,
+    pub claim_signature: Option<HashedUri>,
+
+    pub version: usize,
+}
+
+impl Serialize for Ingredient {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self.version {
+            1 => self.serialize_v1(serializer),
+            2 => self.serialize_v2(serializer),
+            3 => self.serialize_v3(serializer),
+            _ => Err(serde::ser::Error::custom("Ingredient serializer not found")),
```

Comment:
Suggest for this either (a) making `self.version` an enum with the three allowed versions specifically enumerated (and thus impossible to get into this error situation), or (b) making the error message state that an unrecognized version was found.